lib/mocha.js

```diff
@@ -204,6 +204,7 @@ Mocha.prototype.ui = function (name) {
     exports.before = context.before || context.suiteSetup;
     exports.describe = context.describe || context.suite;
     exports.it = context.it || context.test;
+    exports.xit = context.xit || context.test;
```

Comment:
(**This is the only change that should be needed to merge this.** Any others suggested are optional.)

So, the `context.test` on the right side of the `||` isn't right, that's the same as for the non-skipped `it`. (With that said, you can skip to the very bottom for recommended fix; the rest here is explanation.) I'm not 100% certain whether this involves the QUnit interface at all, but it seems these are basically enabling `require`-based used of either the BDD or the TDD interface when either the BDD or TDD interface is used; so, the `||` here would be to replicate the BDD interface in the event the TDD interface is chosen. So, I double-checked what the TDD interface does for skipping.

In the BDD interface, `xit`, `xspecify` and `it.skip` are all set to a function that just calls `it` while only using the title. In the TDD interface, there are no `x` functions, and `test.skip` delegates to the "common interface" `test.skip`, which in turn relies on the TDD interface's `test` function and -- like the BDD function -- just passes it only the title. (Weird coupling between the "common interface" and the TDD interface... but nothing worth worrying about here.)

*So it looks like it should be perfectly safe to change this to `|| context.test.skip`.*

Reply:
@ScottFreeCode  Updated the PR with suggested changes. This time not updated mocha.js at root level.